### DISCUSSION TITLE
Reason for ports not showing up in Klayout

### DISCUSSION THREAD

**Discussion by gmgopikrishnan** [gdsfactory/gdsfactory#1279] [Q&A] [answered]
_Hi
I am just starting up with the gdsfactory layout tutorials. I am trying to run this code:__

import gdsfactory as gf

@gf.cell
def straight(length=10, width=1, layer=(1, 0)):
    WG = gf.Component()
    WG.add_polygon([(0, 0), (length, 0), (length, width), (0, width)], layer=layer)
    WG.add_port(
        name="o1", center=[0, width / 2], width=width, orientation=180, layer=layer
    )
    WG.add_port(
        name="o2", center=[length, width / 2], width=width, orientation=0, layer=layer
    )
    return WG

c = gf.Component("straights_not_connected")

wg1 = c << straight(length=6, width=2.5, layer=1)
wg2 = c << straight(length=11, width=2.5, layer=2)
wg3 = c << straight(length=15, width=2.5, layer=3)

wg2.movey(10).rotate(10)
wg3.movey(20).rotate(15)
wg2.connect("o1", wg1.ports["o2"])
wg3.connect("o1", wg2.ports["o2"])

c.show(show_ports=True, show_subports=True, port_marker_layer=(1,10))
gdspath = c.write_gds_with_metadata("Test_001.GDS")

_But can get only the components to show in Klayout. The ports of the components like "wg1", "wg2", "wg3" are not showing up. Nor is it showing up in the "Test_001.GDS" file into which I try writing.
Could you let me know if I am doing something wrong?_

**Comment by joamatab** [comment, marked as answer]
Hi, the ports are not stored when saving to gds by default 

If you use write_gds_with_metadata    
You can store them in YAML

Another option is to add pins before writing the gds

**Reply by gmgopikrishnan** [reply]
_Hi Joamatab_
_Thanks for the answer! I add the lines:_

gf.add_pins.add_pin_triangle(component=WG, port=WG.ports["o1"], layer = (10,1))
gf.add_pins.add_pin_triangle(component=WG, port=WG.ports["o2"], layer = (10,1))

_and the pins showed up in the gds file.
Thanks!_
